pkg/publicapi/endpoint/compute/endpoint_approve.go

```diff
@@ -20,27 +20,25 @@ import (
 //	@Failure	403	{object}	string
 //	@Failure	500	{object}	string
 //	@Router		/api/v1/compute/approve [get]
-func (s *Endpoint) approve(res http.ResponseWriter, req *http.Request) {
-	request, err := signatures.UnmarshalSigned[bidstrategy.ModerateJobRequest](req.Context(), req.Body)
+func (s *Endpoint) approve(c echo.Context) error {
+	request, err := signatures.UnmarshalSigned[bidstrategy.ModerateJobRequest](c.Request().Context(), c.Request().Body)
 	if err != nil {
-		publicapi.HTTPError(req.Context(), res, err, http.StatusBadRequest)
-		return
+		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
 	}
 
 	approvingClient := os.Getenv("BACALHAU_JOB_APPROVER")
 	if request.ClientID != approvingClient {
-		err := errors.New("approval submitted by unknown client")
-		publicapi.HTTPError(req.Context(), res, err, http.StatusUnauthorized)
-		return
+		return echo.NewHTTPError(http.StatusUnauthorized, errors.New("approval submitted by unknown client"))
 	}
 
-	executions, err := s.store.GetExecutions(req.Context(), request.JobID)
+	executions, err := s.store.GetExecutions(c.Request().Context(), request.JobID)
 	if err != nil {
-		publicapi.HTTPError(req.Context(), res, err, http.StatusInternalServerError)
-		return
+		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
 	}
 
 	for _, execution := range executions {
-		go s.bidder.ReturnBidResult(req.Context(), execution, &request.Response)
+		go s.bidder.ReturnBidResult(c.Request().Context(), execution, &request.Response)
```

Comment:
I don't think we want to use the request context for this routine, as I believe the context will end before the bid result can be returned (might be a bug, although I realize the behavior wasn't added here). Since we have been using the in-memory job store that doesn't respect context cancelation (unlike the new Bolt store) it has probably gone unnoticed. But the change @rossjones is making in https://github.com/bacalhau-project/bacalhau/pull/2779 might cause this to surface at some point.

Do we even need the go-routine here, the BaseEndpoint may fail to update all execution states, in which case we incorrectly(?) signal to the user that the job has been approved when in fact it may be in a bad state. Perhaps this should be made synchronous, our modified use a waitgroup allowing parallelism, but avoiding a premature return or canceled request context.

Reply:
I don't think we need to worry much about `approve` APIs at the moment as we've removed that path in the requester's side. I created a backlog item to remove this https://expanso.atlassian.net/browse/ENG-86